src/osd/OSD.cc

```diff
@@ -6287,6 +6293,50 @@ int OSD::_do_command(
     probe_smart(devid, ds);
   }
 
+  else if (prefix == "drop cache") {
+    dout(20) << "clearing all caches" << dendl;
+    // Clear the objectstore's cache - onode for Bluestore, pagecache for Filestore
+    r = store->flush_cache(&ss);
+    if (r < 0) {
+      ds << "Error flushing objectstore cache: " << cpp_strerror(r);
+      goto out;
+    }
+    // Clear osd map cache
+    {
+      Mutex::Locker l(service.map_cache_lock);
+      service.map_cache.clear();
+    }
+    // Clear the objectcontext cache (per PG)
+    vector<PGRef> pgs;
+    _get_pgs(&pgs);
+    for (auto& pg: pgs) {
+      pg->clear_cache();
+    }
+  }
+
+  else if (prefix == "get_cache_object_count") {
+    int store_cache_count = store->get_cache_obj_count();
+    int obj_ctx_count = 0;
+    int osd_map_count = service.map_cache.get_count();
+    vector<PGRef> pgs;
+    _get_pgs(&pgs);
+    for (auto& pg: pgs) {
+      obj_ctx_count += pg->get_cache_obj_count();
+    }
+    if (f) {
+      f->open_object_section("caches_object_count");
+      f->dump_int("object_ctx", obj_ctx_count);
+      f->dump_int("objectstore_onode", store_cache_count);
+      f->dump_int("osd_map", osd_map_count);
```

Comment:
@gregsfortytwo should we keep printing the stats of the osdmap cache here?

Reply:
I guess as a counterpart to "drop cache" we probably shouldn't? And I think the number of cached osd maps is exposed as a perfcounter anyway